### DISCUSSION TITLE
MikroORM relation field with id

### DISCUSSION THREAD

**Discussion by bicho19** [mikro-orm/mikro-orm#6168] [General]
### Describe the bug

I have a an entity named `Deployment` with a `many-to-one` relation with `Application`. and a application: Rel<Application> field in the Deployment entity. 

My issue is when I fetch a single deployment record, Typescript infer the application field of type Application although it is coming as string. Is there any way to add let's say application_id field within the Deployment entity to fix this ? Because when i tired to add it manually i get : Duplicate fieldNames are not allowed. 

### Reproduction

[https://gist.github.com/bicho19/e54606dfe540b8c04c1e0863ec83f40c](https://gist.github.com/bicho19/e54606dfe540b8c04c1e0863ec83f40c)

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.3.6

### Node.js version

20

### Operating system

MacOS

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Entity definition is not a [reproduction](https://github.com/mikro-orm/reproduction).

> My issue is when I fetch a single deployment record

So share what you are doing, not just how your entity looks like.

> 6.3.6

And always first upgrade to latest version before reporting anything.
